libfreerdp/core/transport.c

```diff
@@ -59,17 +59,130 @@
 
 static void* transport_client_thread(void* arg);
 
+#ifdef WITH_KRB5
+
+#include <krb5.h>
+#include <winpr/library.h>
+static UINT32 transport_krb5_check_account(char* username, char* domain, char* passwd)
+{
+	krb5_error_code ret;
+	krb5_context context;
+	krb5_principal principal = NULL;
+	char address[256];
+	krb5_ccache ccache;
+	krb5_init_creds_context ctx = NULL;
+	_snprintf(address, sizeof(address), "%s@%s", username, domain);
+
+	/* Create a krb5 library context */
+	if ((ret = krb5_init_context(&context)) != 0)
+		WLog_ERR(TAG, "krb5_init_context failed with error %d", (int)ret);
+	else if ((ret = krb5_parse_name_flags(context, address, 0, &principal)) != 0)
+		WLog_ERR(TAG, "krb5_parse_name_flags failed with error %d", (int)ret);
+	/* Find a credential cache with a specified client principal */
+	else if ((ret = krb5_cc_cache_match(context, principal, &ccache)) != 0)
+	{
+		if ((ret = krb5_cc_default(context, &ccache)) != 0)
+			WLog_ERR(TAG, "krb5 failed to resolve credentials cache with error %d", (int)ret);
+	}
+
+	if (ret != KRB5KDC_ERR_NONE)
+		goto out;
+	/* Create a context for acquiring initial credentials */
+	else if ((ret = krb5_init_creds_init(context, principal, NULL, NULL, 0, NULL, &ctx)) != 0)
+	{
+		WLog_WARN(TAG, "krb5_init_creds_init returned error %d", (int)ret);
+		goto out;
+	}
+	/* Set a password for acquiring initial credentials */
+	else if ((ret = krb5_init_creds_set_password(context, ctx, passwd)) != 0)
+	{
+		WLog_WARN(TAG, "krb5_init_creds_set_password returned error %d", ret);
+		goto out;
+	}
+
+	/* Acquire credentials using an initial credential context */
+	ret = krb5_init_creds_get(context, ctx);
+out:
+
+	switch (ret)
+	{
+		case KRB5KDC_ERR_NONE:
+			break;
+
+		case KRB5_KDC_UNREACH:
```

Comment:
KRB5_KDC_UNREACH shouldn't be in this case. In my opinion, it should return a specific message in WLog_WARN indicating it cannot contact the Kerberos server (you can keep the same error FREERDP_ERROR_AUTHENTICATION_FAILED)